pulp_python/app/tasks/repair.py

```diff
@@ -82,6 +97,85 @@ def repair_metadata(content: QuerySet[PythonPackageContent]) -> int:
                 batch = []
                 set_of_update_fields.clear()
 
+        # For on-demand content, we expect that:
+        # 1. PythonPackageContent always has correct name and version, and one ContentArtifact
+        # 2. RemoteArtifact always has correct sha256
+        # Repair is only supported if all PythonPackageContent items with the same name and
+        # version (i.e. group) share the same remote URL. Otherwise, the entire group is skipped
+        for (name, version), group in groupby(
+            on_demand_content.iterator(chunk_size=1000),
+            key=lambda x: (x.name, x.version),
+        ):
+            group = list(group)
+            remotes = set(
+                remote
+                for content in group
+                for remote in content.contentartifact_set.get()
+                .remoteartifact_set.all()
+                .values_list("remote__url", flat=True)
+            )
+            if len(remotes) != 1:
+                log.warning(
+                    _("Only one remote url is supported for {} {}").format(
+                        name, version
```

Comment:
Is enforcing the same `Remote.url` correct?. I will rework this part either way, as iterating through `group` twice and using  `list(group)` is not efficient, but it would be good to know how the code should behave before doing so.

Reply:
Solved, this is wrong